src/native/macos.rs

```diff
@@ -1009,45 +1187,76 @@ where
 
     let () = msg_send![window, makeKeyAndOrderFront: nil];
 
-    let ns_app: ObjcId = msg_send![class!(NSApplication), sharedApplication];
+    struct SendHack<F>(F);
+    unsafe impl<F> Send for SendHack<F> {}
 
-    // Basically reimplementing msg_send![ns_app, run] here
-    let () = msg_send![ns_app, finishLaunching];
+    let state = SendHack(state_original.clone());
+    thread::spawn(move || {
```

Comment:
Hey, somehow I ended up here again :) 
I runned into this discussion: https://hero.handmade.network/forums/code-discussion/t/1409-main_game_loop_on_os_x#7886, and this made me question, do we really need this thread::spawn and [nsapp run] later? If we process the messages, maybe we can process them on right main thread?

Reply:
Yeah, this was my previous approach
https://github.com/not-fl3/miniquad/pull/443
And I thought that I'm smart that I found it
Because it was my first experience of programming for osx and I found how to do window loop even though apple recommends using timer approach for games
But it seems that it isn't this simple and introduces multiple bugs, that should've been taken in account
https://github.com/not-fl3/miniquad/issues/455
https://github.com/not-fl3/miniquad/pull/464
And there would be more of them I think
So I decided to not fix bugs that I cannot predict and used a simple way

As I understand, sokol-app uses a timer approach and [NSApplication run]
https://github.com/floooh/sokol/blob/master/sokol_app.h#L3582

But, raylib via glfw uses nextEventMatchingMask loop and I can try to analyze their code and implement it as before, but possibly with less bugs
https://github.com/glfw/glfw/blob/master/src/cocoa_window.m#L1523